### ISSUE TITLE
Shifted Fourier Bessel Coefficients

### ISSUE THREAD

**Issue by chris-langfield** [ComputationalCryoEM/ASPIRE-Python#601]
Some ideas here:

Relevant Math Stackexchange post: https://math.stackexchange.com/questions/236010/integral-of-scaled-bessel-function-with-linear-phase

![Screen Shot 2022-03-22 at 12 03 42 PM](https://user-images.githubusercontent.com/34426450/159524537-d9654648-b0b7-43cb-ad9f-aaf753e485a3.png)

**Comment by chris-langfield** [comment]
Should have linked the paper: https://arxiv.org/pdf/1412.0781.pdf

Starting from Equation 16

**Comment by janden** [comment]
I'm not sure that the formula for F(f') in the second equality is correct here. There's a dependence on the angle θ as well. See

![image](https://user-images.githubusercontent.com/3976052/161070477-4b6a5d94-2249-4c2e-b178-7901ec19b8b4.png)

[Rangan et al. (2020)](https://arxiv.org/pdf/1905.12317.pdf). Here that would amount to

F(f')(r, θ) = e^{i ξ r₀ cos(θ₀ - θ)} F(f)(r, θ)

You may want to look through that paper. They use a slightly different version of FB coefficients, but some of the calculations might carry over.

**Comment by chris-langfield** [comment]
Yes you're absolutely right. I think it makes the rest of my blurb incorrect. Thanks for the paper. I also want to drop this one in this discussion for the future because it was useful to me

N. Baddour, 2D Fourier transforms in Polar Coordinates. Invited paper, Advances in Imaging and Electron Physics, Volume 165, Editor: Peter Hawkes, Burlington, Academic Press, 2011, pp. 001-046. ISSN 1076-5670, dx.doi.org/10.1016/B978-0-12-385861-0.00001-4.

**Comment by garrettwrong** [comment]
I think that this would be worth doing, but the main place we use shifting I think that we are already in real space where we can shift easily (by pixels at least).  Might be an okay back burner project.

@j-c-c encountered some shifting in the common lines code.  If those are in FB then this might have reason to implement (would need some idea of compute time savings for that problem..)

**Comment by garrettwrong** [comment]
Following up, the shifts in CL are PolarFT, so unrelated to this.

However, it is likely we will be revisiting the class alignment code for performance at some point.  Currently that is crudely shifting by pixels.  I might make more sense to perform in the basis and at a sub pixel level.

**Comment by garrettwrong** [comment]
@oscarmickelin
